widget/embedded/src/components/Quotes/Quotes.tsx

```diff
@@ -0,0 +1,96 @@
+import { useAppStore } from '../../store/AppStore';
+import { useQuoteStore } from '../../store/quote';
+import { QuoteErrorType, type SelectedQuote } from '../../types';
+import { generateQuoteWarnings } from '../../utils/quote';
+import { NoResult } from '../NoResult';
+import { QuoteSkeleton } from '../QuoteSkeleton';
+
+import { ErrorsContainer } from './Quotes.styles';
+
+const ITEM_SKELETON_COUNT = 3;
+export function Quotes(props: PropTypes) {
+  const { loading, onClickOnQuote, fetch } = props;
+  const {
+    selectedQuote,
+    quotes,
+    updateQuotePartialState,
+    fromToken,
+    toToken,
+    error,
+  } = useQuoteStore();
+  const { slippage, customSlippage } = useAppStore();
+  const tokens = useAppStore().tokens();
+  const hasQuotes = !!quotes && quotes.results.length;
+  const userSlippage = customSlippage ?? slippage;
+  const getQuoteWarning = (quote: MultiRouteSimulationResult) => {
+    if (!fromToken || !toToken || !quotes) {
+      return null;
+    }
+    const mergedQuote: SelectedQuote = {
+      requestAmount: quotes.requestAmount, // Assuming quotes is an array
+      ...quote,
+    };
+
+    return generateQuoteWarnings(mergedQuote, {
+      fromToken,
+      toToken,
+      userSlippage,
+      tokens,
+    });
+  };
+  const showNoResultMessage =
+    error?.type === QuoteErrorType.NO_RESULT ||
+    error?.type === QuoteErrorType.REQUEST_FAILED;
+  return loading ? (
+    Array.from({ length: ITEM_SKELETON_COUNT }, (_, index) => (
+      <React.Fragment key={index}>
+        <QuoteSkeleton tagHidden={false} type="list-item" expanded={false} />
+        <Divider size={16} />
+      </React.Fragment>
+    ))
+  ) : (
+    <>
+      {hasQuotes
+        ? quotes.results?.map((quote) => {
+            const quoteWarning = getQuoteWarning(quote);
+            const quoteError = getQuoteError(quote.swaps);
+
+            return (
+              <React.Fragment key={quote.requestId}>
+                <QuoteInfo
+                  selected={selectedQuote?.requestId === quote.requestId}
+                  tagHidden={false}
+                  quote={{ ...quote, requestAmount: quotes.requestAmount }}
+                  loading={loading}
+                  error={quoteError?.options || null}
+                  warning={quoteWarning}
+                  onClick={(quote) => {
+                    updateQuotePartialState('warning', quoteWarning);
+                    updateQuotePartialState(
+                      'error',
+                      quoteError?.options || null
+                    );
+                    onClickOnQuote(quote);
+                  }}
+                  type="list-item"
+                />
+                <Divider size={16} />
+              </React.Fragment>
+            );
+          })
+        : showNoResultMessage && (
+            <ErrorsContainer>
+              <NoResult error={error} fetch={fetch} />
+            </ErrorsContainer>
+          )}
+    </>
```

Comment:
I believe splitting this into two statements would enhance its readability:
{loading && <> ... </>}
{!loading && <> ... </>}

Reply:
fixed